tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/UsageReport/CatalogFileEntry.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.SourceBuild.Tasks.UsageReport
+{
+    internal class CatalogFileEntry
+    {
+        const string ElementName = "File";
+
+        internal string Path { get; set; }
+        internal byte[] OriginalHash { get; set; }
+        internal byte[] PoisonedHash { get; set; }
+
+        public XElement ToXml() => new XElement(ElementName,
+            new XAttribute(nameof(Path), Path),
+            new XAttribute(nameof(OriginalHash), OriginalHash.ToHexString()),
+            PoisonedHash == null ? null : new XAttribute(nameof(PoisonedHash), PoisonedHash.ToHexString())
```

Comment:
Use `ToXAttributeIfNotNull` on `PoisonedHash?.ToHexString()`?
https://github.com/dotnet/source-build/blob/79b70dc0f353d2e652d83ee91936a7d74dce3cae/tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/UsageReport/XmlParsingHelpers.cs#L21-L22

Reply:
Moved this out of the assembly so this is not available anymore.